### PR TITLE
Add timezone documentation for Painless datetimes

### PR DESCRIPTION
**Pull request by jdconrad**
This adds a section to the Painless datetime documentation that describes how to change timezone for string datetimes and complex datetimes.

### CODE REVIEW

File: docs/painless/painless-guide/painless-datetime.asciidoc [hunk 1]
```diff
@@ -335,6 +335,41 @@ if (zdt1.isAfter(zdt2)) {
 }
 ----
 
+==== Datetime Zone
+
+Both string datetimes and complex datetimes have a modifiable timezone with a
```
Comment (rjernst) [inline docs/painless/painless-guide/painless-datetime.asciidoc]: The timezone isn't acutally modifiable, calling withX methods creates a new ZonedDateTime. For this reason, we also shouldn't use the term `change` below. Perhaps reword this to indicate a new ZDT is created?

File: docs/painless/painless-guide/painless-datetime.asciidoc [hunk 2]
```diff
@@ -544,7 +579,8 @@ indexed documents
 +
 [source,Painless]
 ----
-if (doc.containsKey('start_datetime') && doc.containsKey('end_datetime')) { <1>
+if (doc.containsKey('start_datetime') &&
+        doc.containsKey('end_datetime')) { <1>
```
Comment (rjernst) [inline docs/painless/painless-guide/painless-datetime.asciidoc]: what's the reason for the newline? can it at least be aligned with the opening paren if a newline is necessary?

**Review by rjernst — APPROVED** [review]
Looks good just a couple minor comments

File: docs/painless/painless-guide/painless-datetime.asciidoc (hunk 1 above)
Comment (jdconrad) [reply docs/painless/painless-guide/painless-datetime.asciidoc]: Good catch.  Fixed and added a note about immutability.

File: docs/painless/painless-guide/painless-datetime.asciidoc (hunk 2 above)
Comment (jdconrad) [reply docs/painless/painless-guide/painless-datetime.asciidoc]: The <1> was getting cut off, and the code snippets are frustrating to read when scrolling horizontally.  I will re-align this.

Comment (jdconrad) [reply docs/painless/painless-guide/painless-datetime.asciidoc]: I actually shortened the names a bit to make this all fit on one line as it's far easier to read that way.

**Comment by jdconrad** [comment]
@rjernst Thanks for the review! Will commit as soon as CI passes.

**Comment by jdconrad** [comment]
@elasticmachine run elasticsearch-ci/1
